### ISSUE TITLE
Crash and error loop on app launch

### ISSUE THREAD

**Issue by rolandsdrktower** [mihonapp/mihon#363]
### Steps to reproduce

On app launch, the app loops and am unable to share crash logs or restart.

https://github.com/mihonapp/mihon/assets/6371063/1b82ed6b-c2f4-4634-9a16-fa7c1969f2eb

### Expected behavior

App launch normally

### Actual behavior

App crashes and loops

### Crash logs

![Screenshot_20240207-164812](https://github.com/mihonapp/mihon/assets/6371063/a666530f-ad30-4cbd-b337-a8e1c07fb7a2)

### Mihon version

0.16.3 and Preview r6501

### Android version

Android 14 QPR2 and QPR3

### Device

Google Pixel 7

### Other details

The app was working until this latest version.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by MajorTanya** [comment]
Known issue with Google's Android 14 Beta 3.1/QPR2/3, currently no solution exists and considering other A14 devices aren't facing this issue, we have so far assumed it's Google's fault and that they messed up somewhere. I'm leaving this open for now for when we find a solution for this or we find out Google fixed their A14 beta.

**Comment by etihwjd** [comment]
So I've been wrangling with this issue since Tachiyomi went down. I've been using TachiyomiSY and as of 1.10.1 the issue isn't in that fork, but once it resourced to Mihon, the issue has arisen in TachSY too. So while the issue maybe a result of changes in A14 QPR beta 3

**Comment by MajorTanya** [comment]
> So I've been wrangling with this issue since Tachiyomi went down. I've been using TachiyomiSY and as of 1.10.1 the issue isn't in that fork, but once it resourced to Mihon, the issue has arisen in TachSY too. So while the issue maybe a result of changes in A14 QPR beta 3

Even Google knows about this, there is an issue about this exact error in their issue tracker.

**Comment by MajorTanya** [comment]
Because of the dependencies being updated, the issue lies somewhere in how Google's A14 QPRs interact wrongly with some Compose versions

**Comment by etihwjd** [comment]
The dependencies in Mihon, or A14? If it's Mihon, could there be a roll back to the version used in TachSY until the A14 issue is fixed? IDK whether you have telemetry or data indicating what percentage of users run QPRs to know if the change would be worthwhile, but I want my toons! 🤣

**Comment by etihwjd** [comment]
> Only Pixel Beta users seem to be affected, bit extreme to ask an app to adhere to everything a beta can break. You enrolled in the beta knowing it might break stuff, it's a beta. It's also bad practice to fix issues caused by something being faulty, better fix the faulty thing than make everything think faulty is good. If the issue is there in the stable release it will get fixed, until then it's probably easier for you to roll back to an earlier version of an app that works.

Entirely understandable, and rereading my comment it looks like I'm asking for a rollback, but I was asking _if_ there could be a roll back. As in if there was a change in Mihon's code, _could_ this be rolled back if there was a necessity to do so. But I understand the logic of "_if it ain't broke, don't fix it_", and if it's a fault with Google, they should fix it. 👍 I'm not a software dev, so I don't understand the underlying causes but understand that as a beta user I should give feedback and bug reports, which is what I was trying to do here. Thanks for explaining it for me!

Because I don't know what commit changed the dependencies, could you advise me on the last "good" version that won't death loop with errors like an starving crocodile?! 🤣

**Comment by skiman6010** [comment]
version 0.16.2 works.

**Comment by etihwjd** [comment]
> version 0.16.2 works.

Thanks! 😁 I'm assuming this version is based on stable Tach, is there a pre-change version of Mihon based on Tach beta before the end? I'm a sucker for punishment and preferred the feature set! 🤪

**Comment by MajorTanya** [comment]
The feature set is literally identical, except for a few fixes, what in the hells are you talking about

**Comment by etihwjd** [comment]
> The feature set is literally identical, except for a few fixes, what in the hells are you talking about

Chill. I'm trying to actually be helpful and understand what's going on. I wouldn't be here on github if I didn't actually want to engage with the developers and help, and if I'm asking these questions, no doubt there are other QPR users who have similar questions; answering my questions will help other people who end up here through googling because they're in the same position that I am. Instead of flaming me, you might want to actually engage with me to help the community at large. 🙄 What's the point of being on github if you don't actually want to constructively engage?

The point I'm making is the last time I tested Tach stable there was a notable difference in UI with the beta version, so I decided to carry on using the beta. I'm trying to find a Mihon version that is based on the beta code from tach, prior to the commits that introduced the dependency change, so I can keep on using that until Google sort their shit out.

**Comment by MajorTanya** [comment]
Tachiyomi stable released 0.15.x versions shortly before stopping development early this year. It is possible you missed those versions if you weren't updating as they came out, but that doesn't change anything about the fact Mihon is literally continuing with the same code base that Tachiyomi had.

If you want old Tachiyomi, this isn't the place to ask for it. Believe me when I say I am severely frustrated by Google's absolute faux pax with this issue. And because their corporate internal git isn't public, I cannot even go to check what the fix is, let alone how long it will take.
Every few days, another Pixel 5-8 user in the Android 14 QPR cycles comes into the support Discord server and asks for help with this. Every time, we tell them to downgrade to 0.16.2, which is the last known good version on these beta-enrolled devices.

If we had a magic fix, I would have pushed for it to be included in a new release already, just to not have to look on completely powerlessly any more and tell people to downgrade to a less ideal version.

TL;DR:
1. This is on Google's QPR of A14 (Samsung devices with A14-based OS don't have this issue)
2. Mihon has no silver bullet available
3. The dependency change happened between 0.16.2 and 0.16.3, mostly to solve other issues iirc (which you could deduce from getting told that 0.16.2 works)

**Comment by etihwjd** [comment]
> Mihon 0.16.2 is no more Tachi than Mihon 0.16.3. the code they share with Tachi is the same code they got when the app was created at 0.16.0. [...] Tachi preview and Tachi stable were literally the same at 0.15.3.

Thank you. These two sentences answer the question I was trying to answer. 

> You use these terms that you don't really have a grasp on and keep adding emotes so you sound loony.

Using emojis to add emotional valence to a message may not be the done thing on here, but I'm using them to try and express concepts that are absent in plain text, so forgive me for trying to effectively communicate.  Also, ad homenin attacks, calling someone a looney, reflect more on your style of interaction than using emojis does on myself; it's a great way to add fuel to the smoldering embers of a misunderstanding, so if that's your aim, crack on! 👍



> Tachiyomi stable released 0.15.x versions shortly before stopping development early this year. It is possible you missed those versions if you weren't updating as they came out, but that doesn't change anything about the fact Mihon is literally continuing with the same code base that Tachiyomi had.
> 
> If you want old Tachiyomi, this isn't the place to ask for it. Believe me when I say I am severely frustrated by Google's absolute faux pax with this issue. And because their corporate internal git isn't public, I cannot even go to check what the fix is, let alone how long it will take. Every few days, another Pixel 5-8 user in the Android 14 QPR cycles comes into the support Discord server and asks for help with this. Every time, we tell them to downgrade to 0.16.2, which is the last known good version on these beta-enrolled devices.
> 
> If we had a magic fix, I would have pushed for it to be included in a new release already, just to not have to look on completely powerlessly any more and tell people to downgrade to a less ideal version.
> 
> TL;DR:
> 
> 1. This is on Google's QPR of A14 (Samsung devices with A14-based OS don't have this issue)
> 2. Mihon has no silver bullet available
> 3. The dependency change happened between 0.16.2 and 0.16.3, mostly to solve other issues iirc (which you could deduce from getting told that 0.16.2 works)
> 
> EDIT: https://issuetracker.google.com/issues/322402917

This helps me to understand what's going on. After finding when the change was introduced in stable, I was trying to the last Mihon beta that was based on Mihon 0.16.2, before it was updated to 0.16.3. so I could stay on that until the next A14 QPR beta. I've now figured how to do this, but not when the beta was changed from 0.16.2 to 0.16.3.

I didn't mean to start a massive conflagration, so feel free to get rid of the rubbish in this thread. When I find out when beta moved from 0.16.2 code, I'll drop it here so others can find it. 👍

**Comment by etihwjd** [comment]
So, coming back to the thing that I actually came here to discuss, from what I can see the last version of the beta that is based on 0.16.2 code is **[r6480](https://github.com/mihonapp/mihon-preview/releases/tag/r6480)**. I'm not planning on upgrading the app until I've got a new version of the QPR beta, and I wouldn't suggest anyone doing the same if they want to stay on the Mihon beta channel without having to faff around.

If users are having the restart loop issue, and want to stick on the Mihon beta channel, they should **downgrade to [r6480](https://github.com/mihonapp/mihon-preview/releases/tag/r6480)**, restore their most recent backup, and **not upgrade** the app until there's a new QPR beta.

**Comment by rolandsdrktower** [comment]
Just downloaded QPR 3 beta 2 and did not get crashes on opening latest version of Mohon. Maybe Google fixed their bug?

**Comment by MajorTanya** [comment]
> Just downloaded QPR 3 beta 2 and did not get crashes on opening latest version of Mihon. Maybe Google fixed their bug?

@rolandsdrktower are you using the stable or preview release of Mihon for this? Would be good to know for other users with this A14 QPR mess.

**Comment by rolandsdrktower** [comment]
Tested in stable 0.16.3 and 0.16.4.  Neither have crashed on me yet.

**Comment by MajorTanya** [comment]
And what build number does your current system have?

**Comment by rolandsdrktower** [comment]
No worries!  I tested first in Mihon 0.16.4, then uninstalled and tested again in Mihon 0.16.3 (just to rule out a change in 0.16.4 fixing the issue).  Both worked fine.  I'm currently using TachiyomiSY 1.10.5 from [SyncYomi](https://github.com/SyncYomi/SyncYomi) dev which is based on Mihon 0.16.4, and I'm not seeing the same behavior I initially reported.

**Comment by zenmaster3000** [comment]
As the debugging expert that I am, I discovered that's if you restart your phone, you will be able to opem the Mihon app afterwards.

> App version: 0.16.4 (standard, a4f5dfa, 5, 2024-02-26T16:26Z)
> Android version: 14 (SDK 34; build AP1A.240305.019.A1)
> Device brand: google
> Device manufacturer: Google
> Device name: cheetah (cheetah)
> Device model: Pixel 7 Pro
> WebView: Android System WebView 122.0.6261.119

**Comment by Lookitzmike** [comment]
> As the debugging expert that I am, I discovered that's if you restart your phone, you will be able to open the Mihon app afterwards.
> 
> > App version: 0.16.4 (standard, [a4f5dfa](https://github.com/mihonapp/mihon/commit/a4f5dfab1a874f4875d3bbfd67472a6803e7a7cf), 5, 2024-02-26T16:26Z)
> > Android version: 14 (SDK 34; build AP1A.240305.019.A1)
> > Device brand: google
> > Device manufacturer: Google
> > Device name: cheetah (cheetah)
> > Device model: Pixel 7 Pro
> > WebView: Android System WebView 122.0.6261.119

Can confirm, this worked.

**Comment by rdtx2015** [comment]
> > As the debugging expert that I am, I discovered that's if you restart your phone, you will be able to open the Mihon app afterwards.
> > > App version: 0.16.4 (standard, [a4f5dfa](https://github.com/mihonapp/mihon/commit/a4f5dfab1a874f4875d3bbfd67472a6803e7a7cf), 5, 2024-02-26T16:26Z)
> > > Android version: 14 (SDK 34; build AP1A.240305.019.A1)
> > > Device brand: google
> > > Device manufacturer: Google
> > > Device name: cheetah (cheetah)
> > > Device model: Pixel 7 Pro
> > > WebView: Android System WebView 122.0.6261.119
> 
> Can confirm, this worked.

Worked for me too but after a few hours the cycle crash happens again.

Pixel 8

**Comment by kokounet** [comment]
I have the same issue on my pixel 6
Android version number: AP1A.240305.019.A1
Rebooting can fix the issue temporarily but it comes back at some point

**Comment by Dreiundachzig** [comment]
I'm using Pixel 8 Pro with official Android 14 (no Beta version) and just had this problem.

**Comment by nathanhoel** [comment]
> > I'm using Pixel 8 Pro with official Android 14 (no Beta version) and just had this problem.
> 
> Google pushed it out to their normal OS updates, so it's not just Beta OS now

It appears that way. Pixel 6 Pro,  not on beta, same issue now.

**Comment by MajorTanya** [comment]
This should be solved now with the 0.16.5 stable release of Mihon. If you're on an earlier version of Mihon and experiencing this, please update your app to the latest version.

Should you experience this issue again with the new version of Mihon (0.16.5 or newer), please reach out again.

Closing as fixed.
